tests/fixtures/expect.go

```diff
@@ -34,38 +37,117 @@ type Expect struct {
 	kubeClient    kubernetes.Interface
 	gitSync       *v1alpha1.GitSync
 	gitSyncClient planepkg.GitSyncInterface
+	currentCommit string
 }
 
-// check that resources are created
-func (e *Expect) ResourcesExist(resourceType string, resources []string) *Expect {
+// check that resources are created / exist
+func (e *Expect) ResourcesExist(apiVersion, resourceType string, resources []string) *Expect {
 
 	e.t.Helper()
-	ctx := context.Background()
+	e.t.Log("Verifying that resources exist..")
+
 	for _, r := range resources {
-		result := e.kubeClient.CoreV1().RESTClient().Get().
-			Timeout(defaultTimeout).
-			Namespace(TargetNamespace).
-			Resource(resourceType).
-			Name(r).
-			Do(ctx)
-		if result.Error() != nil {
-			e.t.Fatalf("Resource %s does not exist", r)
+		if !e.doesExist(apiVersion, resourceType, r) {
+			e.t.Fatalf("Resource %s/%s does not exist", resourceType, r)
 		}
+		e.t.Logf("Resource %s/%s exists", resourceType, r)
 	}
 
 	return e
 }
 
-// check that resources are deleted
-func (e *Expect) ResourcesDontExist(resourceType string, resources []string) *Expect {
+// check that resources have been deleted or never existed
+func (e *Expect) ResourcesDontExist(apiVersion, resourceType string, resources []string) *Expect {
 
 	e.t.Helper()
+	e.t.Log("Verifying that resources no longer exist..")
+
 	for _, r := range resources {
-		if !e.isDeleted(resourceType, r) {
-			e.t.Fatalf("Resource %s not deleted", r)
+		if !e.doesNotExist(apiVersion, resourceType, r) {
+			e.t.Fatalf("Resource %s/%s not deleted", resourceType, r)
+		}
+		e.t.Logf("Resource %s/%s doesn't exist", resourceType, r)
+	}
+
+	return e
+}
+
+// verify value of resource spec to determine if change occurred or not as expected
+func (e *Expect) VerifyResourceState(apiVersion, resourceType, resource, field, key string, value interface{}) *Expect {
+
+	e.t.Helper()
+	e.t.Log("Verifying resource state is as expected..")
+
+	var (
+		err  error
+		body []byte
+	)
+
+	if !e.doesExist(apiVersion, resourceType, resource) {
+		e.t.Fatalf("Resource %s/%s does not exist", resourceType, resource)
+	}
+
+	ctx := context.Background()
+
+	if apiVersion == "v1" {
+		body, err = e.kubeClient.CoreV1().RESTClient().
+			Get().
+			Namespace(TargetNamespace).
+			Resource(resourceType).
+			Name(resource).
+			DoRaw(ctx)
+	} else {
+		body, err = e.kubeClient.CoreV1().RESTClient().Get().AbsPath(
+			fmt.Sprintf("/apis/%s/namespaces/%s/%s/%s",
+				apiVersion,
+				TargetNamespace,
+				resourceType,
+				resource)).DoRaw(ctx)
```

Comment:
Just curious - does this call not work for CoreV1 Resources?

Reply:
Yes, I wasn't able to `Get` the example configMaps or secrets I was creating.

Reply:
okay, thanks